### PR TITLE
Support to configure the maximum number of displayed items

### PR DESCRIPTION
**Pull request by horber**
image
image

### CODE REVIEW

**Comment by Fine0830** [comment]
Make scene to me. BTW, you'd better mention issue first. After discussion, you pull a request and this avoids multiple modifications.

**Comment by horber** [comment]
> Make sense to me. BTW, you'd better mention issue first. After discussion, you pull a request and this avoids multiple modifications.

**Comment by horber** [comment]
Sorry, I just accidentally turned off this PR

**Comment by horber** [comment]
> Make sense to me. BTW, you'd better mention issue first. After discussion, you pull a request and this avoids multiple modifications.

👌

**Comment by wu-sheng** [comment]
Feature makes sense to me. @Fine0830 Your decision on codes.

**Comment by wu-sheng** [comment]
And, we should not update all existing templates, please make sure the default value is there and as same as the previous version.

File: src/views/components/dashboard/charts/chart-edit.vue [hunk 1]
```diff
@@ -237,6 +237,16 @@ limitations under the License. -->
           @change="setItemConfig({ type: 'height', value: $event.target.value })"
         />
       </div>
+      <div class="flex-h mb-5" v-show="!isChartType">
+        <div class="title grey sm">{{ $t('maxItemNum') }}:</div>
+        <input
+          type="number"
+          min="1"
+          class="rk-chart-edit-input long"
+          :value="itemConfig.maxItemNum"
```
Comment (Fine0830) [inline src/views/components/dashboard/charts/chart-edit.vue:246]: Please set the default value of `maxItemNum` to `10`.

**Review by Fine0830 — CHANGES_REQUESTED** [review]

File: src/views/components/dashboard/charts/chart-edit.vue (hunk 1 above)
Comment (horber) [reply src/views/components/dashboard/charts/chart-edit.vue:246]: I feel that all the default data is configured on the back end through the 'queryGetAllTemplates' API, There may be a problem with setting the default value at the front end. 
image

Comment (wu-sheng) [reply src/views/components/dashboard/charts/chart-edit.vue:246]: The default works when this config missing. Which are really missing for all configuration.

File: src/views/components/dashboard/charts/chart-edit.vue [hunk 2]
```diff
@@ -347,14 +350,24 @@ limitations under the License. -->
...
       if (!this.itemConfig.independentSelector || this.pageTypes.includes(this.type)) {
         return;
       }
       this.setItemServices();
     }
 
+    private setDefaultValue(itemConfig: any) {
+      const keys: string[] = Object.keys(itemConfig);
+      let key: string = '';
+      for (key in this.itemConfigDefault) {
```
Comment (Fine0830) [inline src/views/components/dashboard/charts/chart-edit.vue:364]: I think `for...of` is better than `for ..in`. Could you update this?

File: src/views/components/dashboard/charts/chart-edit.vue (hunk 1 above)
Comment (horber) [reply src/views/components/dashboard/charts/chart-edit.vue:246]: ok, It makes sense

**Comment by Fine0830** [comment]
Please resolve conflicts.

File: src/views/components/dashboard/charts/chart-edit.vue (hunk 2 above)
Comment (horber) [reply src/views/components/dashboard/charts/chart-edit.vue:364]: why?

Comment (Fine0830) [reply src/views/components/dashboard/charts/chart-edit.vue:373]: The approach has some downside. https://stackoverflow.com/questions/500504/why-is-using-for-in-for-array-iteration-a-bad-idea

Comment (horber) [reply src/views/components/dashboard/charts/chart-edit.vue]: got it,  already update. Is that ok?

**Comment by wu-sheng** [comment]
@Fine0830 Please take a look when you have time.

**Review by Fine0830 — APPROVED** [review]
LGTM

**Pull request merged**